src/control/common/file_utils.go

```diff
@@ -238,26 +229,53 @@ func Run(cmd string) error {
 	return err
 }
 
-// FindBinary attempts to locate the named binary by checking
-// $PATH first. If the binary is not found in $PATH, then
-// it looks in the directory containing the binary for
-// the running process.
+// GetWorkingPath retrieves path relative to the current working directory when
+// invoking the current process.
+func GetWorkingPath(inPath string) (string, error) {
+	if path.IsAbs(inPath) {
+		return inPath, nil
+	}
+
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to determine working directory")
+	}
+
+	return path.Join(workingDir, inPath), nil
+}
+
+// GetAdjacentPath retrieves path relative to the binary used to launch the
+// currently running process.
+func GetAdjacentPath(inPath string) (string, error) {
+	if path.IsAbs(inPath) {
+		return inPath, nil
+	}
+
+	selfPath, err := os.Readlink("/proc/self/exe")
+	if err != nil {
+		return "", errors.Wrap(err, "unable to determine path to self")
+	}
+
+	return path.Join(path.Dir(selfPath), inPath), nil
+}
```

Comment:
This doesn't sit right with me. As a control plane developer using our internal APIs, I would have to go read the source of these helper functions to discover that there is a potential plot twist: If the supplied path is an absolute path, then it's returned unmodified. That behavior doesn't match the function names or descriptions.

I would prefer to move the responsibility for checking path.IsAbs() and making a decision about what happens next to the caller. It might be slightly more code, but it's far more readable and easier to reason about.

If you want to make config-specific helpers that encapsulate this logic that would be fine, too. The helpers here in common are intended to be general-purpose and so their behavior should be straightforward and generally applicable.